src/test/java/net/snowflake/client/jdbc/ResultSetVectorLatestIT.java

```diff
@@ -0,0 +1,172 @@
+      enforceQueryResultFormat(stmt);
+      try(ResultSet resultSet = stmt.executeQuery("select null::vector(float, 2)")) {
+        assertTrue(resultSet.next());
+        Integer[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Integer.class);
+        assertNull(result);
+        assertVectorMetadata(resultSet, 1, Types.FLOAT, 1);
+      }
+    }
+  }
+
+  @Test
+  public void testGetIntVectorFromTable() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      stmt.execute("create or replace table test_vector_int(x vector(int, 2), y int)");
+      stmt.execute("insert into test_vector_int select [3, 7]::vector(int, 2), 15");
+      try (ResultSet resultSet = stmt.executeQuery("select x, y from test_vector_int")) {
+        assertTrue(resultSet.next());
+        Integer[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Integer.class);
+        assertEquals(new Integer[]{3, 7}, result);
+        assertVectorMetadata(resultSet, 1, Types.INTEGER, 2);
+      }
+    }
+  }
+
+  @Test
+  public void testGetFloatVectorFromTable() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      enforceQueryResultFormat(stmt);
+      stmt.execute("create or replace table test_vector_float(x vector(float, 2), y float)");
+      stmt.execute("insert into test_vector_float select [-3, 7.1]::vector(float, 2), 20.3");
+      try (ResultSet resultSet = stmt.executeQuery("select x, y from test_vector_float")) {
+        assertTrue(resultSet.next());
+        Float[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Float.class);
+        assertEquals(new Float[]{-3f, 7.1f}, result);
+        assertVectorMetadata(resultSet, 1, Types.FLOAT, 2);
+      }
+    }
+  }
+
+  private <T extends Number> String vectorToString(T[] vector, String vectorType) {
+    return Arrays.toString(vector) + "::vector(" + vectorType + ", " + vector.length + ")";
+  }
+
+  private void enforceQueryResultFormat(Statement stmt) throws SQLException {
+    String sql =
+        String.format(
+            "alter session set jdbc_query_result_format = '%s'", queryResultFormat.toUpperCase());
+    stmt.execute(sql);
+  }
+
+  private void assertVectorMetadata(ResultSet resultSet, int vectorColumnIndex, int expectedVectorFieldType, int allColumns) throws SQLException {
+    ResultSetMetaData metadata = resultSet.getMetaData();
+    assertEquals(allColumns, metadata.getColumnCount());
+    assertEquals(EXTRA_TYPES_VECTOR, metadata.getColumnType(vectorColumnIndex));
+    assertEquals("VECTOR", metadata.getColumnTypeName(vectorColumnIndex));
+    SnowflakeResultSetMetaDataV1 sfMetadata = (SnowflakeResultSetMetaDataV1) metadata;
+    assertTrue(sfMetadata.isStructuredTypeColumn(vectorColumnIndex));
```

Comment:
Is it true indeed? Should we treat vector as structured type? WDYT @sfc-gh-pmotacki ?

Reply:
removed

Reply:
I meant not removing - maybe this function `isStructuredTypeColumn` should return false here? I'll let Przemek to decide.

Reply:
In this context, all types containing a schema with field descriptions are structured types, so the vector should be the same as an array. If I understand correctly, a vector can work without knowing about the fields inside.

Reply:
I restored the assertion cheking that it is a structured type